### PR TITLE
Tile QoL changes More to come in sister pr's

### PR DESCRIPTION
**Pull request by dwasint**
# About The Pull Request
Splitting this into 2+ pr's as things like neon carpets are 100 files changed alone, and tileing 2 is 70
Adds various QoL changes for tiles, like tile sprayer for decals, glass tiles, and added transparency flag for material datums (not currently used)
Also ports over asset caching helps load times.
PRS USED:
https://github.com/tgstation/tgstation/pull/63503
https://github.com/tgstation/tgstation/pull/54184
https://github.com/tgstation/tgstation/pull/54233
https://github.com/tgstation/tgstation/pull/66301
https://github.com/tgstation/tgstation/pull/49402  <-- Only the turf additions
https://github.com/tgstation/tgstation/pull/54250
https://github.com/tgstation/tgstation/pull/62890
https://github.com/tgstation/tgstation/pull/67449
https://github.com/tgstation/tgstation/pull/66248/

## Why It's Good For The Game
More variety for building and mapping

## Testing Photographs and Procedure

<details>
<summary>Screenshots&Videos</summary>

Put screenshots and videos here with an empty line between the screenshots and the `<details>` tags.

</details>

## Changelog

:cl:
add:new glass floor types lets you see space or the z below
add: tile sprayer you can now apply some decals to floors
/:cl:

### CODE REVIEW

File: code/game/turfs/open/space/space.dm [hunk 1]
```diff
@@ -63,6 +63,13 @@
 	if (opacity)
 		has_opaque_atom = TRUE
 
+	var/turf/T = SSmapping.get_turf_above(src)
```
Comment (nednaZ) [inline code/game/turfs/open/space/space.dm:66]: One letter variable

**Review by nednaZ — CHANGES_REQUESTED** [review]
Not seeing anything major.
Will need to test it when I get a chance

**Comment by NanoCats** [comment]
Is it deliberate that you can't print Toner cartridges?

Autolathe or Protolathe on TG IIRC

**Comment by nednaZ** [comment]
Only request is that you do add a method to create or get more of those toner cartridges, otherwise you'll just have to print a new sprayer every time it runs out.

**Review by nednaZ — APPROVED** [review]
Looks good to me.
Tests have shown no issues.

**Pull request merged**